In [2]:
import pandas as pd
import numpy as np
from src import soporte_limpieza as sl
pd.set_option("display.max_columns",None)

>Carga del archivo y análisis visual de las columnas: Esto nos dará un mayo entendimiento del contexto de las columnas  

In [11]:
df = pd.read_pickle("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-unidos.pkl")
df.sample()

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
3793  Ministério da Agricultura, Pecuária e Abastec          22201   

                                          NOMBRE ÓRGANO  \
3793  Instituto Nacional de Colonização e Reforma Ag...   

      CÓDIGO UNIDAD GESTORA                          NOMBRE UNIDAD GESTORA  \
3793                 373001  DPTO. DE ADMINISTRACAO FINANCEIRA-DAF - INCRA   

     CATEGORÍA ECONÓMICA                               ORIGEN INGRESO  \
3793  Receitas Correntes  Impostos, Taxas e Contribuições de Melhoria   

     TIPO INGRESO                                        DETALLE  \
3793        Taxas  TAXAS INSPEÇÃO,CONTROLE E FISCALIZAÇÃO-PRINC.   

      VALOR PREVISTO ACTUALIZADO  VALOR LANZADO  VALOR REALIZADO  \
3793                         0.0            0.0        133928.91   

      PORCENTAJE REALIZADO FECHA LANZAMIENTO  AÑO EJERCICIO  
3793                   0.0        2018-12-14           2018

> Ver el tamaño de una estructura de datos. Esto es de gran utilidad para verificar cuántos datos se cargaron desde un archivo.

In [12]:
df.shape

(1026299, 15)

> Utilizamos el atributo df.columns en pandas para conocer el nombre de todas las columnas de una manera rápida

In [13]:
df.columns

Index(['NOMBRE ÓRGANO SUPERIOR', 'CÓDIGO ÓRGANO', 'NOMBRE ÓRGANO',
       'CÓDIGO UNIDAD GESTORA', 'NOMBRE UNIDAD GESTORA', 'CATEGORÍA ECONÓMICA',
       'ORIGEN INGRESO', 'TIPO INGRESO', 'DETALLE',
       'VALOR PREVISTO ACTUALIZADO', 'VALOR LANZADO', 'VALOR REALIZADO',
       'PORCENTAJE REALIZADO', 'FECHA LANZAMIENTO', 'AÑO EJERCICIO'],
      dtype='object')

> Obtenemos toda la información del DataFrame para analizarlo

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1026299 entries, 0 to 134592
Data columns (total 15 columns):
 #   Column                      Non-Null Count    Dtype         
---  ------                      --------------    -----         
 0   NOMBRE ÓRGANO SUPERIOR      1026299 non-null  object        
 1   CÓDIGO ÓRGANO               1026299 non-null  int64         
 2   NOMBRE ÓRGANO               1026299 non-null  object        
 3   CÓDIGO UNIDAD GESTORA       1026299 non-null  int64         
 4   NOMBRE UNIDAD GESTORA       1026299 non-null  object        
 5   CATEGORÍA ECONÓMICA         1026299 non-null  object        
 6   ORIGEN INGRESO              1026299 non-null  object        
 7   TIPO INGRESO                1026299 non-null  object        
 8   DETALLE                     1026299 non-null  object        
 9   VALOR PREVISTO ACTUALIZADO  1026299 non-null  float64       
 10  VALOR LANZADO               1026299 non-null  float64       
 11  VALOR REALIZADO             10

**Insights:** 

- El DataFrame contiene 1026299 registros y 15 columnas con información sobre la ejecución de ingresos públicos en Brasil durante los años 2013 al 2021.

- Todas las columnas tienen datos completos después de trabajar sobre los datos nulos que nos dio la columna "FECHA DE LANZAMIENTO".

- Los tipos de datos incluyen 2 columnas numéricas enteras (int64), 7 categóricas (object), 4 numérica flotante (float64) y 2 de fecha (datetime64[ns] e int32). 

- El DataFrame ocupa aproximadamente 121.4 MB en memoria.

> Analisis descriptivo: Análisis estadístico básico, organizado en filas para facilitar la lectura, comparación y diagnóstico inicial de las columnas de un DataFrame.

In [17]:
df_descriptivos_numericos = df.describe().T
df_descriptivos_numericos

count                           mean  \
CÓDIGO ÓRGANO               1026299.0                   31435.161341   
CÓDIGO UNIDAD GESTORA       1026299.0                  231293.743317   
VALOR PREVISTO ACTUALIZADO  1026299.0                27611421.049683   
VALOR LANZADO               1026299.0                 5242806.106963   
VALOR REALIZADO             1026299.0                24980422.309138   
PORCENTAJE REALIZADO        1026299.0                      80.313812   
FECHA LANZAMIENTO             1026299  2018-09-20 14:26:49.968756480   
AÑO EJERCICIO               1026299.0                    2018.225555   

                                            min                  25%  \
CÓDIGO ÓRGANO                           20101.0              25201.0   
CÓDIGO UNIDAD GESTORA                  110005.0             154003.0   
VALOR PREVISTO ACTUALIZADO         -214773743.0                  0.0   
VALOR LANZADO              -148347267246.910004                  0.0   
VALOR REALIZADO            -156285934188.029999              260.905   
PORCENTAJE REALIZADO                  -114552.0                  0.0   
FECHA LANZAMIENTO           2013-12-31 00:00:00  2017-03-31 00:00:00   
AÑO EJERCICIO                            2013.0               2017.0   

                                            50%                  75%  \
CÓDIGO ÓRGANO                           26298.0              36000.0   
CÓDIGO UNIDAD GESTORA                  170013.0             250088.0   
VALOR PREVISTO ACTUALIZADO                  0.0                  0.0   
VALOR LANZADO                               0.0                  0.0   
VALOR REALIZADO                         3022.32            40853.855   
PORCENTAJE REALIZADO                        0.0                  0.0   
FECHA LANZAMIENTO           2018-08-27 00:00:00  2020-02-12 00:00:00   
AÑO EJERCICIO                            2018.0               2020.0   

                                            max                std  
CÓDIGO ÓRGANO                           91214.0       10382.647574  
CÓDIGO UNIDAD GESTORA                  913001.0      144178.420681  
VALOR PREVISTO ACTUALIZADO      1603521711208.0  3393367175.127452  
VALOR LANZADO               357160677863.969971   810883695.355811  
VALOR REALIZADO             771117711060.949951    1406892548.8405  
PORCENTAJE REALIZADO                 72363772.0       71481.112902  
FECHA LANZAMIENTO           2021-12-03 00:00:00                NaN  
AÑO EJERCICIO                            2021.0           1.737966

**Insights:** 


1.	CÓDIGO ÓRGANO: Código numérico que identifica la entidad gubernamental superior.

    - Al tratarse de una columna con códigos que identifican la entidad gubernamental superior, no podemos sacar conclusiones relevantes del análisis descriptivo.

2.	CÓDIGO UNIDAD GESTORA: Código numérico de la unidad gestora responsable.

	- De igual manera que en el caso anterior, no podemos no podemos sacar conclusiones relevantes del análisis descriptivo.

3.	VALOR PREVISTO ACTUALIZADO: Monto actualizado del ingreso previsto (formato texto).

	- Podemos observar que los percentiles son 0, lo que quiere decir que la mayoría de los datos son 0. Lo cual quiere decir que el valor previsto actualizado en la mayoría de los casos es 0.
	- El valor mínimo es negativo y con una cantidad bastante grande (-214.773.743), algo que parece extraño al ser un "valor previsto actualizado"
	- El valor máximo (1.603.521.711.208) es muy alto si lo comparamos con la media (27.611.321). Además, si lo comparamos con la desviación estándar (3.393.367.175), vemos que hay una gran dispersión en los datos. Es decir, hay datos muy altos y muy bajos, por lo que el valor máximo es algo puntual.

4.	VALOR LANZADO: Monto que fue registrado como lanzado

	- Vemos que nos sucede algo similar al anterior caso. Los percentiles son 0, lo cuál quiere decir que la mayoría de los datos son 0.
	- El valor mínimo (-148.347.267.246) es negativo, lo cuál es extraño al ser un valor lanzado. 
	- Si comparamos la media (5.242.806) con el valor máximo (357.160.677.863) hay mucha diferencia y, viendo la desviación estándar (810.883.695) podemos confirmar que hay uan gran desviación de los datos. Por lo que hay mucha diferencia entre unos valores y otros.

5.	VALOR REALIZADO: Monto realmente recaudado

	- El 25% de los datos están por encima de 260,905, el 50% (3.022,32) refleja que la mitad de los registros tienen ese valor o menos y el 75% indica que el 75% de los registros tiene valores por dejabjo de 408.53,855. Por lo que podemos afirmar que en la mayoría de los casos los registros tienden a tener valores pequeños.
	- El valor mínimo es de -156.285.934.188,029999, algo bastante extraño al tratarse de de cantidades recaudadas.
	- El valor máximo es demasiado alto en comparación con el promedio y la desviación estándar nos dice que hay mucha dispersión de los datos.

6.	PORCENTAJE REALIZADO: Porcentaje de ejecución respecto al valor previsto.

	- El principal dato a destacar es que hay datos en negativo, lo cual quiere decir que la ejecución del valor está por debajo del valor previsto.
	- El valor promedio de 80,3% quiere decir que en la mayoría de los casos el porcentaje de ejecución es bastante alto.
	- Gracias a los percentiles podemos ver que la mayoría de datos son 0.
	- La desviación estándar nos dice que hay mucha dispersión de datos, lo cuál quiere decir que hay porcentajes muy altos y muy bajos.

7.	FECHA LANZAMIENTO: Fecha en la que se registró la ejecución del ingreso.

	- El valor mínimo (2013-12-31) lo cuál quiere decir que la primera fecha de la base de datos es del 31 de diciembre de 2013 y la última del 3 de diciembre de 2021.

8.	AÑO EJERCICIO: Año correspondiente a la ejecución de los ingresos.

	- 

Conclusión:


Los datos muestran una alta variabilidad, con valores tanto negativos como cercanos a cero, lo cuál es llamativo al tratarse de ingresos, es decir, los ingresos registrados tienen una gran dispersión en relación al valor previsto y lanzado. Por lo que hay que realizar un análisis de los datos para saber qué está pasando.
La ejecución promedio respecto al valor previsto es alta (80%), aunque muchos registros tienen un porcentaje de ejecución de 0%, lo que refleja que una parte significativa de los ingresos aún no se ha realizado o ejecutado.

In [21]:
df_descriptivos_object = df.describe(include = "O").T
df_descriptivos_object

count unique  \
NOMBRE ÓRGANO SUPERIOR  1026299     25   
NOMBRE ÓRGANO           1026299    287   
NOMBRE UNIDAD GESTORA   1026299    356   
CATEGORÍA ECONÓMICA     1026299      5   
ORIGEN INGRESO          1026299     15   
TIPO INGRESO            1026299     63   
DETALLE                 1026299   1886   

                                                                      top  \
NOMBRE ÓRGANO SUPERIOR                             Ministério da Educação   
NOMBRE ÓRGANO           Ministério da Economia - Unidades com vínculo ...   
NOMBRE UNIDAD GESTORA             SETORIAL ORCAMENTARIA E FINANCEIRA / ME   
CATEGORÍA ECONÓMICA                                    Receitas Correntes   
ORIGEN INGRESO                                  Outras Receitas Correntes   
TIPO INGRESO                 Serviços Administrativos e Comerciais Gerais   
DETALLE                       SERV.ADMINISTRAT.E COMERCIAIS GERAIS-PRINC.   

                          freq  
NOMBRE ÓRGANO SUPERIOR  361891  
NOMBRE ÓRGANO           119811  
NOMBRE UNIDAD GESTORA   108481  
CATEGORÍA ECONÓMICA     979596  
ORIGEN INGRESO          322907  
TIPO INGRESO            278157  
DETALLE                 159039

**Insights:** 
Análisis conciso de los datos:

1. NOMBRE ÓRGANO SUPERIOR: Nombre de la entidad gubernamental superior.

	- De los 1260 registros disponibles, hay 5 niveles educativos únicos, siendo el más frecuente Bachelor con 479 empleados.

2.	NOMBRE ÓRGANO: Nombre de la entidad gubernamental específica.
	– Los empleados pertenecen a 6 campos educativos distintos, predominando Life Sciences con 606 empleados.

3.	NOMBRE UNIDAD GESTORA: Nombre de la unidad gestora.
	- Hay solo 2 géneros reportados, con una mayor representación masculina (Male) en 882 empleados.

4.	CATEGORÍA ECONÓMICA: Clasificación económica de los ingresos (por ejemplo, "Receitas Correntes").
	- Los empleados se dividen en 3 estados civiles, siendo el más común Married con 673 empleados.

5.	ORIGEN INGRESO: Fuente específica del ingreso (por ejemplo, "Outras Receitas Correntes").
	- Cada empleado tiene un identificador único, lo que asegura 1470 valores únicos en esta columna.

6.	TIPO INGRESO: Tipo de ingreso dentro de la fuente (por ejemplo, "Demais receitas correntes").
	- De los 1161 registros disponibles, todos los empleados están marcados como mayores de 18 años (Y), mostrando ausencia de variabilidad en esta categoría.

7.	DETALLE: Detalle adicional del tipo de ingreso (por ejemplo, "Receita de honorários de advogados").
	- Los empleados provienen de 70 países, siendo Noruega el más frecuente con 46 empleados.

Conclusión:

Estos datos describen características cualitativas del personal. La mayoría de los empleados tienen títulos de Bachelor, pertenecen al campo de Life Sciences, y predominan los hombres (Male) casados (Married). 

Existe una alta satisfacción en dimensiones como el ambiente laboral (EnvironmentSatisfaction) y las relaciones laborales (RelationshipSatisfaction), así como un alto nivel de involucramiento laboral y rendimiento excelente. 

La homogeneidad en Over18 sugiere poca utilidad analítica en esa variable, mientras que la diversidad en países y educación podría ser relevante para análisis más profundos.